core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1239,7 +1233,7 @@ public SearchSourceBuilder readFrom(StreamInput in) throws IOException {
             builder.queryBuilder = in.readQuery();
         }
         if (in.readBoolean()) {
-            int size = in.readVInt();
+            size = in.readVInt();
```

Comment:
This (and the following occurences) looks like its overwriting the global size field of the SearchSourceBuilder now. That's not the builder that is returned, so it might not matter but I find it very confusing. So I'd suggest also renaming the local variable to avoid shadowing the global name.